app/src/main/java/br/com/stonesdk/sdkdemo/activities/MifareActivity.kt

```diff
@@ -0,0 +1,323 @@
+package br.com.stonesdk.sdkdemo.activities
+
+import android.content.DialogInterface
+import android.os.Bundle
+import android.text.InputType
+import android.view.View
+import android.widget.TextView
+import android.widget.Toast
+import androidx.appcompat.app.AlertDialog
+import androidx.appcompat.app.AppCompatActivity
+import br.com.stone.posandroid.hal.api.mifare.MifareKeyType
+import br.com.stone.posandroid.providers.PosMifareProvider
+import br.com.stonesdk.sdkdemo.R
+import br.com.stonesdk.sdkdemo.databinding.ActivityMifareBinding
+import br.com.stonesdk.sdkdemo.databinding.DualInputDialogBinding
+import br.com.stonesdk.sdkdemo.databinding.InputDialogBinding
+import stone.application.interfaces.StoneCallbackInterface
+
+/**
+ * https://sdkandroid.stone.com.br/reference/provider-de-mifare
+ */
+class MifareActivity : AppCompatActivity() {
+
+    private lateinit var binding: ActivityMifareBinding
+
+    private var logTextView: TextView? = null
+    private var mifareProvider: PosMifareProvider? = null
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+
+        binding = ActivityMifareBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        logTextView = binding.logTextView
+        setCardUUIDText("")
+    }
+
+
+    override fun onStop() {
+        super.onStop()
+        if (mifareProvider != null) mifareProvider!!.cancelDetection()
+    }
+
+    /**
+     * Exemplo de detecção de cartão, pegando o UUID.
+     */
+    fun detectCard(view: View?) {
+        mifareProvider = PosMifareProvider(applicationContext)
+
+        mifareProvider!!.connectionCallback = object : StoneCallbackInterface {
+            override fun onSuccess() {
+                setCardUUIDText(mifareProvider!!.cardUUID.contentToString())
+                logTextView!!.append("Cartão detectado: " + mifareProvider!!.cardUUID.contentToString() + "\n")
+            }
+
+            override fun onError() {
+                runOnUiThread {
+                    Toast.makeText(
+                        this@MifareActivity,
+                        "Erro na detecção",
+                        Toast.LENGTH_SHORT
+                    ).show()
+                    logTextView!!.append(mifareProvider!!.listOfErrors.toString() + "\n")
+                }
+            }
+        }
+        mifareProvider!!.execute()
+    }
+
+    fun cancelDetectCard(view: View?) {
+        if (mifareProvider != null) mifareProvider!!.cancelDetection()
+    }
+
+    /**
+     * Exemplo de leitura do valor de um bloco.
+     * @param sector Número do setor.
+     * @param block Número do bloco relativo ao setor (mínimo 0, máximo 3).
+     * @param key Chave de autenticação do bloco.
+     */
+    private fun executeBlockRead(sector: Int, block: Int, key: ByteArray) {
+        val mifareProvider = PosMifareProvider(applicationContext)
```

Comment:
Aqui não pode ser val. No onStop precisamos da instancia do mifareProvider para cancelar a detecção.
```suggestion
        mifareProvider = PosMifareProvider(applicationContext)
```

Reply:
Commitar a sugestão no github vai quebrar o build. Precisa alterar mais coisa pra lidar com o mifareProvider nullable.